src/types.ts

```diff
@@ -0,0 +1,46 @@
+import { DocumentNode } from 'graphql/language/ast';
+
+export interface ApolloFetcher {
+  request (operation: Operation): Observable;
+}
+
+export interface Operation {
+  query: DocumentNode;
+  variables?: object;
+  operationName?: string;
+  context?: object;
+}
+
+export interface Observable {
+    subscribe(subscriber: Subscriber<FetchResult>): UnsubscribeHandler;
+    subscribe(
+      next: (result: FetchResult) => void,
+      error?: (error: any) => void,
+      complete?: () => void,
+    ): UnsubscribeHandler;
+    status(): object;
+}
+
+export interface Subscriber<T> {
+  next: (result: T) => void;
+  error?: (error: any) => void;
+  complete?: () => void;
+}
+
+export interface FetchResult {
+  data: any;
+  errors?: any;
+  extensions?: any;
+  context?: object;
+}
+
+export type UnsubscribeHandler = () => void;
```

Comment:
I think this is now different in the TC39 proposal. Do we want to try to align here?